CONTRIBUTING.md

```diff
@@ -100,6 +100,24 @@ $ git push fork feature
 
 Open a pull request against the main `opentelemetry-python` repo.
 
+Pull requests are also tested for their compatibility with packages distributed 
+by OpenTelemetry in the [OpenTelemetry Python Contrib Repository](https://github.com/open-telemetry/opentelemetry-python.git).
+
+If a pull request introduces a change that would break the compatibility of
+these packages with the Core packages in this repo, a separate PR should be
+opened in the Contrib repo with changes to make the packages compatible.
+
+Once the pull request in the Contrib repo is open, take note of the commit SHA
+and edit the `CONTRIB_SHA` variable found in `.github/workflows/test.yml` to
+pass even the Contrib repo tests and so get the Core repo PR green and ready to
+merge. Once the Core repo PR has the right `CONTRIB_SHA`, it can be merged.
```

Comment:
You're saying the core `CONTRIB_SHA` should point to a contrib PR when it is merged? That doesn't seem right because everyone else's tests would still point there, even after that contrib PR is merged

Reply:
Sorry I guess I'm getting confused between this PR and https://github.com/open-telemetry/opentelemetry-python/pull/1357

Correct me if I'm wrong but I think that's what we want? We want everyone's tests to point to that Contrib SHA because that Contrib SHA is going to be merged and become the new Contrib REPO master branch SHA. Then as soon as everyone rebases on the merged Core repo they will pass the tests too.

Reply:
I'm going to update this based on the discussion from #1357 . Core points to `master` and Contrib points to a specific SHA. If we decide it should be the reverse in that PR I'll reverse the instructions here again.

Reply:
Hmm I guess it is fine for now. I think your suggestion to just get this in and iterate is a good one :)